app/models/supermarket.rb

```diff
@@ -1,3 +1,10 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
+  def unique_customers
+    Customer.joins(:customer_items)
```

Comment:
Nice job using multiple joins. You could also add an association
```ruby
has_many :customers, through: :items
```
Then you could just start right at `customers`. 
```ruby 
customers.distinct
```